unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/PublicChatChannelController.cs

```diff
@@ -160,12 +165,17 @@ private bool IsOldPrivateMessage(ChatMessage message)
         return timestampInSeconds < initTimeInSeconds;
     }
 
-    private void HandleMessageReceived(ChatMessage message)
+    private void HandleMessageReceived(ChatMessage message) { HandleMessageReceived(message, true); }
+
+    private void HandleMessageReceived(ChatMessage message, bool sendAnalytics)
     {
         if (IsOldPrivateMessage(message)) return;
 
         chatHudController.AddChatMessage(message, View.IsActive);
-        
+
+        if (sendAnalytics && message.sender != ownProfile.userId)
+            socialAnalytics.SendChannelMessageReceived(message.sender, message.body.Length, message.messageType == ChatMessage.Type.SYSTEM ? String.Empty : View.channel, message.messageType);
```

Comment:
responsability issues: the controller should solve the channelId instead of asking it to the view.. this way you can delete `view.Channel` property

Reply:
You're right! I didn't notice that we already had the `channelId` in the controller. Thanks! Done in https://github.com/decentraland/unity-renderer/pull/2257/commits/eced3e36420d66ba4891f9910b191b650a005f07